crates/sui-core/src/epoch/data_removal.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use mysten_metrics::spawn_monitored_task;
+use narwhal_config::Epoch;
+use std::fs;
+use std::path::PathBuf;
+use tokio::sync::mpsc;
+
+pub struct EpochDataRemover {
+    base_path: PathBuf,
+    tx_remove: mpsc::Sender<Epoch>,
+}
+
+impl EpochDataRemover {
+    pub fn new(base_path: PathBuf) -> Self {
+        let (tx_remove, _rx_remove) = mpsc::channel(1);
+        Self {
+            base_path,
+            tx_remove,
+        }
+    }
+
+    pub async fn run(&mut self) {
+        let (tx_remove, mut rx_remove) = mpsc::channel(1);
```

Comment:
As the data truncate is not that critical, maybe we can increase the capacity a little bit? Ex 10?